packages/typespec-ts/test/modularUnit/scenarios/models/deserialization/extends.md

```diff
@@ -719,3 +719,201 @@ export enum KnownVersions {
   _20240801Preview = "2024-08-01-preview",
 }
 ```
+
+# discriminated union with @clientName
+
+## TypeSpec
+
+```tsp
+@doc("Base type for normalizers.")
+@discriminator("@odata.type")
+model LexicalNormalizer {
+  #suppress "@azure-tools/typespec-azure-core/no-string-discriminator" "Existing"
+  #suppress "@azure-tools/typespec-azure-core/casing-style" "Maintain compatibility with existing models."
+  `@odata.type`: string;
+  name: string;
+}
+union TokenFilterName {
+  string,
+  ArabicNormalization: "arabic_normalization",
+}
+union CharFilterName {
+  string,
+  HtmlStrip: "html_strip",
+}
+model CustomNormalizer extends LexicalNormalizer {
+
+  tokenFilters?: TokenFilterName[];
+
+  charFilters?: CharFilterName[];
+
+  #suppress "@azure-tools/typespec-azure-core/casing-style" "Maintain compatibility with existing models."
+  `@odata.type`: "#Microsoft.Azure.Search.CustomNormalizer";
+}
+@@clientName(LexicalNormalizer.`@odata.type`, "odataType");
```

Comment:
using `@@clientName` affected the inheritance of properties with the same name, causing duplication.